File: android/src/main/java/xyz/justzappit/zappmessaging/models/ZMConversation.kt
package xyz.justzappit.zappmessaging.models

import kotlinx.serialization.Serializable

/**
 * A chat conversation (direct or group).
 * Mirrors iOS ZMConversation for cross-platform consistency.
 */
@Serializable
data class ZMConversation(
    /** Unique conversation identifier */
    val id: String,
    /** Type of conversation */
    val type: ConversationType,
    /** Participant public keys (excluding self for groups) */
    val participantIds: List<String>,
    /** Group ID (for group chats only) */
    val groupId: String? = null,
    /** Creator's public key (for groups only) */
    val creatorKey: String? = null,
    /** Display name for the conversation */
    val displayName: String,
    /** Preview of last message */
    val lastMessage: String? = null,
    /** Timestamp of last message (epoch millis) */
    val lastMessageTimestamp: Long? = null,
    /** When the conversation was created (epoch millis) */
    val createdAt: Long = System.currentTimeMillis(),
    /** Whether current user is the group owner */
    val isOwner: Boolean? = null,
    /** Number of unread messages */
    val unreadCount: Int? = null,
    /** Set when the group owner removed us: history stays, nothing more flows. Epoch millis. */
    val removedAt: Long? = null
)

File: ios/Sources/ZappMessaging/Models/ZMConversation.swift
//
//  ZMConversation.swift
//  ZappMessaging
//
//  Represents a direct or group chat conversation
//

import Foundation

/// A chat conversation
public struct ZMConversation: Codable, Identifiable, Equatable, Sendable {
    /// Unique conversation identifier
    public let id: String
    
    /// Type of conversation
    public let type: ConversationType
    
    /// Participant public keys (excluding self for groups)
    public let participantIds: [String]
    
    /// Group ID (for group chats only)
    public let groupId: String?
    
    /// Creator's public key (for groups only)
    public let creatorKey: String?
    
    /// Display name for the conversation
    public var displayName: String
    
    /// Preview of last message
    public var lastMessage: String?
    
    /// Timestamp of last message
    public var lastMessageTimestamp: Date?
    
    /// When the conversation was created
    public let createdAt: Date
    
    /// Whether current user is the group owner
    public let isOwner: Bool?
    
    public init(
        id: String,
        type: ConversationType,
        participantIds: [String],
        groupId: String? = nil,
        creatorKey: String? = nil,
        displayName: String,
        lastMessage: String? = nil,
        lastMessageTimestamp: Date? = nil,
        createdAt: Date = Date(),
        isOwner: Bool? = nil
    ) {
        self.id = id
        self.type = type
        self.participantIds = participantIds
        self.groupId = groupId
        self.creatorKey = creatorKey
        self.displayName = displayName
        self.lastMessage = lastMessage
        self.lastMessageTimestamp = lastMessageTimestamp
        self.createdAt = createdAt
        self.isOwner = isOwner
    }
}

/// Type of conversation
public enum ConversationType: String, Codable, Equatable, Sendable {
    case direct
    case group
}
